### PR TITLE
Get rid of old DNS resolver + improvements (Part 2)

### LINKED ISSUE #71: DnsResolver uses hardcoded DNS server
Hi,

I am using Xamarin Forms and recently updated with latest Nuget package, it works fine in simulator but on connection it is crashing when I run on physical device.

Thanks

### PR DESCRIPTION
**Pull request by vitalyster**
* Removed hardcoded DNS server
* Add workaround for Android
* Verified that it works on iOS without a fix

Closes #71.

### CODE REVIEW

**Comment by ForNeVeR** [comment]
So, we'll have to recommend our Xamarin users to resolve the servers themselves before starting a connection? Is that okay?

File: SharpXMPP.Shared/XmppTcpConnection.cs [hunk 1]
```diff
@@ -26,10 +26,10 @@ public class XmppTcpConnection : XmppConnection
         private TcpClient _client;
 
         /// <summary>
-        /// A list of name servers to use for the DNS lookup. Uses 1.1.1.1 by default. If empty, then will use the name
-        /// servers configured by the local network adapter(s).
+        /// A list of nameservers to use for the DNS lookup. If empty, then will use
+        /// the nameservers configured by the local network adapter(s).
         /// </summary>
-        public IPAddress[] NameServers { get; set; } = { IPAddress.Parse("1.1.1.1") };
+        public static IPAddress[] NameServers { get; set; } = { };
```
Comment (ForNeVeR) [inline SharpXMPP.Shared/XmppTcpConnection.cs:32]: I am concerned with using the static array here. Statics are generally bad.

What would you think of another design: still allowing to do quick static override, but allowing to set the servers per client of required?

```csharp
public static IPAddress[] DefaultNameservers { get; set; } = {};
public IPAddress Nameservers { get; set; } = DefaultNameservers;
```

So, we could populate from default list by default, but still allow per client override (and thus avoid any static-related problems in cases when it's important).

**Comment by vitalyster** [comment]
> So, we'll have to recommend our Xamarin users to resolve the servers themselves before starting a connection? Is that okay?

It is a bug in Xamarin, so we can only provide a workaround.

File: SharpXMPP.Shared/XmppTcpConnection.cs (hunk 1 above)
Comment (vitalyster) [reply SharpXMPP.Shared/XmppTcpConnection.cs:32]: We should not allow overriding nameservers in general, this case is only a workaround for Xamarin. Actually, Nameservers property should not be public, let me know if there is some modifier that exists to hide it from users but allow it to override from child package (SharpXMPP.Client.Droid)?

Comment (vitalyster) [reply SharpXMPP.Shared/XmppTcpConnection.cs:32]: Well, that kind of modifier is useless, because a user client will have different package name

Comment (ForNeVeR) [reply SharpXMPP.Shared/XmppTcpConnection.cs:32]: I see. Ok, good to go then.

**Review by ForNeVeR — APPROVED** [review]

**Comment by ForNeVeR** [comment]
Do you want me to release a new NuGet package?

**Pull request merged**